diff --git a/dist/dev.cjs b/dist/dev.cjs
index d615cc449542355d4eeb4626828280788f8e0120..1325aab11ddc82d188ef8146f7d329b127359f3e 100644
--- a/dist/dev.cjs
+++ b/dist/dev.cjs
@@ -1797,3 +1797,4 @@ exports.startTransition = startTransition;
 exports.untrack = untrack;
 exports.useContext = useContext;
 exports.useTransition = useTransition;
+exports.getSuspenseContext = getSuspenseContext;
diff --git a/dist/dev.js b/dist/dev.js
index c171f4e055d9f2b642d34d6dacd90a5a0149df1c..4523d9e605234bb3b016b3a902ad30016a3fff6c 100644
--- a/dist/dev.js
+++ b/dist/dev.js
@@ -1975,5 +1975,6 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 };
diff --git a/dist/server.cjs b/dist/server.cjs
index e0988e36528aa872e7b095ffd88640f67cabf479..1f95ea1d28d5b4595e07bca4b501d0c27a9a1ade 100644
--- a/dist/server.cjs
+++ b/dist/server.cjs
@@ -731,3 +731,4 @@ exports.startTransition = startTransition;
 exports.untrack = untrack;
 exports.useContext = useContext;
 exports.useTransition = useTransition;
+exports.getSuspenseContext = () => SuspenseContext;
diff --git a/dist/server.js b/dist/server.js
index 3a7bf9af5c528ba765004b0bf052583e4596e169..4199bd98be9977aa7625c4b73fda473fc83fff85 100644
--- a/dist/server.js
+++ b/dist/server.js
@@ -717,6 +717,10 @@ function Suspense(props) {
   }, suspenseError);
 }
 
+function getSuspenseContext () {
+  return SuspenseContext;
+}
+
 export {
   $DEVCOMP,
   $PROXY,
@@ -770,5 +774,6 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 };
diff --git a/dist/solid.cjs b/dist/solid.cjs
index 7c905f678bcb91d05d15a5cb434a3b236a70ff68..06873a62468f2353bba0df65498ad81693f0dc1c 100644
--- a/dist/solid.cjs
+++ b/dist/solid.cjs
@@ -1726,3 +1726,4 @@ exports.startTransition = startTransition;
 exports.untrack = untrack;
 exports.useContext = useContext;
 exports.useTransition = useTransition;
+exports.getSuspenseContext = getSuspenseContext;
diff --git a/dist/solid.js b/dist/solid.js
index 2781f8c71945eddee1025921db29c7d5d25cec99..5298fff32674b2296b394f381d295ade35b078d1 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1873,5 +1873,6 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 };
diff --git a/types/index.d.ts b/types/index.d.ts
index e93ca31f7a0db0e0aa0185cc86b155d903806e05..b0b942b15cee04c3ad2ff09d9c8b1be9a21a850f 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -29,7 +29,8 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 } from "./reactive/signal.js";
 export type {
   Accessor,
