diff --git a/dist/dev.cjs b/dist/dev.cjs
index ca8fceced373d52401970b2ee09f5b1dba73558a..85c817ffbfbe28bb6209922e67341ca1d8e96fcc 100644
--- a/dist/dev.cjs
+++ b/dist/dev.cjs
@@ -1797,3 +1797,4 @@ exports.startTransition = startTransition;
 exports.untrack = untrack;
 exports.useContext = useContext;
 exports.useTransition = useTransition;
+exports.getSuspenseContext = getSuspenseContext;
diff --git a/dist/dev.js b/dist/dev.js
index 7025e77204dd84185d399e1cd6413865ef8dc98d..8fe4ed26ee9ea47670c78d1a731d8d6971d55df5 100644
--- a/dist/dev.js
+++ b/dist/dev.js
@@ -1975,5 +1975,6 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 };
diff --git a/dist/server.cjs b/dist/server.cjs
index e0988e36528aa872e7b095ffd88640f67cabf479..1f95ea1d28d5b4595e07bca4b501d0c27a9a1ade 100644
--- a/dist/server.cjs
+++ b/dist/server.cjs
@@ -731,3 +731,4 @@ exports.startTransition = startTransition;
 exports.untrack = untrack;
 exports.useContext = useContext;
 exports.useTransition = useTransition;
+exports.getSuspenseContext = () => SuspenseContext;
diff --git a/dist/server.js b/dist/server.js
index 3a7bf9af5c528ba765004b0bf052583e4596e169..4199bd98be9977aa7625c4b73fda473fc83fff85 100644
--- a/dist/server.js
+++ b/dist/server.js
@@ -717,6 +717,10 @@ function Suspense(props) {
   }, suspenseError);
 }
 
+function getSuspenseContext () {
+  return SuspenseContext;
+}
+
 export {
   $DEVCOMP,
   $PROXY,
@@ -770,5 +774,6 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 };
diff --git a/dist/solid.cjs b/dist/solid.cjs
index 6e925743d89555749a7861772fa75671b6d18d7f..445b95872e68c44fc89abc5165c317d5d56e66e7 100644
--- a/dist/solid.cjs
+++ b/dist/solid.cjs
@@ -1726,3 +1726,4 @@ exports.startTransition = startTransition;
 exports.untrack = untrack;
 exports.useContext = useContext;
 exports.useTransition = useTransition;
+exports.getSuspenseContext = getSuspenseContext;
diff --git a/dist/solid.js b/dist/solid.js
index 7cfbb3bf7abc85beab27a3863fc0312bcf4ef500..d80e36822a18757a610ff7c02ee027eb3dbd4f7d 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1534,7 +1534,6 @@ function createUniqueId() {
   return ctx ? `${ctx.id}${ctx.count++}` : `cl-${counter++}`;
 }
 
-const narrowedError = name => `Stale read from <${name}>.`;
 function For(props) {
   const fallback = "fallback" in props && {
     fallback: () => props.fallback
@@ -1564,7 +1563,6 @@ function Show(props) {
                 keyed
                   ? c
                   : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
                       return props.when;
                     }
               )
@@ -1612,7 +1610,6 @@ function Switch(props) {
               keyed
                 ? when
                 : () => {
-                    if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
                     return cond.when;
                   }
             )
@@ -1873,5 +1870,6 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 };
diff --git a/types/index.d.ts b/types/index.d.ts
index e93ca31f7a0db0e0aa0185cc86b155d903806e05..b0b942b15cee04c3ad2ff09d9c8b1be9a21a850f 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -29,7 +29,8 @@ export {
   startTransition,
   untrack,
   useContext,
-  useTransition
+  useTransition,
+  getSuspenseContext
 } from "./reactive/signal.js";
 export type {
   Accessor,
