diff --git a/dist/solid.js b/dist/solid.js
index b9ced44f69369d2db93452cb7fbd9476f411e890..d6bcd84a6d6bf23defee9d51f0adf5c7668950bf 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1581,7 +1581,6 @@ function Show(props) {
                 keyed
                   ? c
                   : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
                       return props.when;
                     }
               )
@@ -1628,7 +1627,6 @@ function Switch(props) {
               keyed
                 ? when
                 : () => {
-                    if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
                     return cond.when;
                   }
             )
