diff --git a/dist/solid.js b/dist/solid.js
index 2571609363d5758d32e2aa88489acca146a6d308..4a1e0bf4e523fb075340e0c2ccef0a4a9ab12f7f 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1581,7 +1581,6 @@ function Show(props) {
                 keyed
                   ? c
                   : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
                       return props.when;
                     }
               )
@@ -1628,7 +1627,6 @@ function Switch(props) {
               keyed
                 ? when
                 : () => {
-                    if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
                     return cond.when;
                   }
             )
