diff --git a/dist/solid.js b/dist/solid.js
index d901eeb90512010f488bc8b2b9cfdb148fcd8031..9da829c1816af4e888466bb1ec029cdeb1c8123b 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1447,7 +1447,6 @@ function Show(props) {
       const child = props.children;
       const fn = typeof child === "function" && child.length > 0;
       return fn ? untrack(() => child(keyed ? c : () => {
-        if (!untrack(condition)) throw narrowedError("Show");
         return props.when;
       })) : child;
     }
@@ -1478,7 +1477,6 @@ function Switch(props) {
     const c = cond.children;
     const fn = typeof c === "function" && c.length > 0;
     return fn ? untrack(() => c(keyed ? when : () => {
-      if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
       return cond.when;
     })) : c;
   }, undefined, undefined);
