diff --git a/dist/solid.js b/dist/solid.js
index 7a5701a42b2cde619ad091e01aafb0fe5eecf709..aa1f6e4459572cf212fd301b3452f8735b435ca5 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1543,7 +1543,6 @@ function createUniqueId() {
   return ctx ? `${ctx.id}${ctx.count++}` : `cl-${counter++}`;
 }
 
-const narrowedError = name => `Stale read from <${name}>.`;
 function For(props) {
   const fallback = "fallback" in props && {
     fallback: () => props.fallback
@@ -1572,10 +1571,7 @@ function Show(props) {
               child(
                 keyed
                   ? c
-                  : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
-                      return props.when;
-                    }
+                  : () => props.when
               )
             )
           : child;
@@ -1619,10 +1615,7 @@ function Switch(props) {
             c(
               keyed
                 ? when
-                : () => {
-                    if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
-                    return cond.when;
-                  }
+                : () => cond.when
             )
           )
         : c;
