diff --git a/dist/solid.js b/dist/solid.js
index 7350fc223921ca774966b22cb77e8542e28b0e25..bfa0e6c2a0056e256fe31b4c8aebade1cdfcce8b 100644
--- a/dist/solid.js
+++ b/dist/solid.js
@@ -1573,7 +1573,6 @@ function Show(props) {
                 keyed
                   ? c
                   : () => {
-                      if (!untrack(condition)) throw narrowedError("Show");
                       return props.when;
                     }
               )
@@ -1620,7 +1619,6 @@ function Switch(props) {
               keyed
                 ? when
                 : () => {
-                    if (untrack(evalConditions)[0] !== index) throw narrowedError("Match");
                     return cond.when;
                   }
             )
