diff --git a/dist/client/build/register.js b/dist/client/build/register.js
index f89bb9eaf7307099d7f019e0fbcdcd215077709a..ea1cb67ee9abaa9dc4cac17c72aab18e24f5e838 100644
--- a/dist/client/build/register.js
+++ b/dist/client/build/register.js
@@ -7,6 +7,7 @@ function registerSW(options = {}) {
   const {
     immediate = false,
     onNeedRefresh,
+    onBeginUpdate,
     onOfflineReady,
     onRegistered,
     onRegisteredSW,
@@ -73,6 +74,12 @@ function registerSW(options = {}) {
         }
       }
       wb.register({ immediate }).then((r) => {
+        if (onBeginUpdate) {
+          r?.addEventListener('updatefound', () => {
+            onBeginUpdate();
+          });
+        }
+
         if (onRegisteredSW)
           onRegisteredSW("__SW__", r);
         else
diff --git a/types/index.d.ts b/types/index.d.ts
index fcb2a5639308f37d6f9a1ed9e16987ddf4a1dac2..d6b1c71d15ba53a8a1cc46cfda4489b44e52b24b 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -1,6 +1,7 @@
 export interface RegisterSWOptions {
   immediate?: boolean
   onNeedRefresh?: () => void
+  onBeginUpdate?: () => void
   onOfflineReady?: () => void
   /**
    * Called only if `onRegisteredSW` is not provided.
