diff --git a/dist/node/chunks/dep-BKbDVx1T.js b/dist/node/chunks/dep-BKbDVx1T.js
index 6d370a34b5d43a57878d9207ec61fffcf631f0ed..2919998eb28db4454aa06deaf9c4167dd7b30e11 100644
--- a/dist/node/chunks/dep-BKbDVx1T.js
+++ b/dist/node/chunks/dep-BKbDVx1T.js
@@ -68459,7 +68459,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
