diff --git a/dist/node/chunks/dep-jvB8WLp9.js b/dist/node/chunks/dep-jvB8WLp9.js
index 2d612f3b8f56d6aec9def07ec765c97024c16a78..d662a1f5ac92e705e796b71d72ca14d2150f6bf7 100644
--- a/dist/node/chunks/dep-jvB8WLp9.js
+++ b/dist/node/chunks/dep-jvB8WLp9.js
@@ -66852,7 +66852,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
