diff --git a/dist/node/chunks/dep-9A4-l-43.js b/dist/node/chunks/dep-9A4-l-43.js
index 990f7c7945986bd26f0b9dac5794b28a235df1c7..4e6691a931619a037ca9137baa0bc71a37c987dd 100644
--- a/dist/node/chunks/dep-9A4-l-43.js
+++ b/dist/node/chunks/dep-9A4-l-43.js
@@ -66991,7 +66991,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
