diff --git a/dist/node/chunks/dep-jDlpJiMN.js b/dist/node/chunks/dep-jDlpJiMN.js
index 741fe7a5e92ebbc1c38eb4044f73079f9bbdaeb4..16e74654e682fa6ead2b2fcc0a4cad77ddf5f53c 100644
--- a/dist/node/chunks/dep-jDlpJiMN.js
+++ b/dist/node/chunks/dep-jDlpJiMN.js
@@ -66740,7 +66740,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
