diff --git a/dist/node/chunks/dep-BEhTnQAI.js b/dist/node/chunks/dep-BEhTnQAI.js
index fc5798dd21e54ed0c709bbbc6d5fb301d3af9cbd..a4cac6a8d3012e26bcf0a157297e0c121f37cc6f 100644
--- a/dist/node/chunks/dep-BEhTnQAI.js
+++ b/dist/node/chunks/dep-BEhTnQAI.js
@@ -65157,7 +65157,7 @@ async function resolveBuildPlugins(config) {
       ...config.isWorker ? [webWorkerPostPlugin()] : []
     ],
     post: [
-      buildImportAnalysisPlugin(config),
+      ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
       ...config.esbuild !== false ? [buildEsbuildPlugin(config)] : [],
       ...options.minify ? [terserPlugin(config)] : [],
       ...!config.isWorker ? [
