diff --git a/dist/node/chunks/dep-stQc5rCc.js b/dist/node/chunks/dep-stQc5rCc.js
index 47f3abd8895308e1e7a194f9c192be879d2c2fcf..b051a66d90cbd2d66c1e6e2df0bbc73a4e6b5bee 100644
--- a/dist/node/chunks/dep-stQc5rCc.js
+++ b/dist/node/chunks/dep-stQc5rCc.js
@@ -66708,7 +66708,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
