diff --git a/dist/node/chunks/dep-94_H5fT6.js b/dist/node/chunks/dep-94_H5fT6.js
index 5c8b0a0f68a580efc1ff2d63f53ad0c1f6526818..3b299680defdef73dc2472f4ded739a1a0fab8d5 100644
--- a/dist/node/chunks/dep-94_H5fT6.js
+++ b/dist/node/chunks/dep-94_H5fT6.js
@@ -66707,7 +66707,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
