diff --git a/dist/node/chunks/dep-V3BH7oO1.js b/dist/node/chunks/dep-V3BH7oO1.js
index 628c10d380dd760b1f5f54baddbaa37747b335d6..a41d45d97728d63e32fbecebadc7eeb143bd5cd8 100644
--- a/dist/node/chunks/dep-V3BH7oO1.js
+++ b/dist/node/chunks/dep-V3BH7oO1.js
@@ -66988,7 +66988,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
