diff --git a/dist/node/chunks/dep-BBHrJRja.js b/dist/node/chunks/dep-BBHrJRja.js
index 340ed927819a9bd92f69ac9193b914fb13ca4bd7..1aec67e7b7d4e1d899a530972153d990f808efbf 100644
--- a/dist/node/chunks/dep-BBHrJRja.js
+++ b/dist/node/chunks/dep-BBHrJRja.js
@@ -67080,7 +67080,7 @@ async function resolveBuildPlugins(config) {
             ...(config.isWorker ? [webWorkerPostPlugin()] : []),
         ],
         post: [
-            buildImportAnalysisPlugin(config),
+            ...(config.build.modulePreload !== false ? [buildImportAnalysisPlugin(config)] : []),
             ...(config.esbuild !== false ? [buildEsbuildPlugin(config)] : []),
             ...(options.minify ? [terserPlugin(config)] : []),
             ...(!config.isWorker
